x-pack/legacy/plugins/maps/public/components/layer_addpanel/source_select/source_select.js

```diff
@@ -24,12 +26,16 @@ export function SourceSelect({
       ? <EuiIcon type={Source.icon} size="l" />
       : null;
 
+    const sourceTitle = Source.isBeta
+      ? <div><span>{Source.title}</span>{generateBetaBadge(Source.title)}</div>
+      : Source.title;
+
     return (
       <Fragment key={Source.type}>
         <EuiSpacer size="s" />
         <EuiCard
```

Comment:
EuiCard has a `betaBadgeLabel` and `betaBadgeTooltipContent` prop where it will apply the Beta badge properly to the card itself without having to add it directly.